client/asset/dcr/dcr.go

```diff
@@ -3536,6 +3537,78 @@ func (dcr *ExchangeWallet) signTxAndAddChange(baseTx *wire.MsgTx, feeRate uint64
 	return msgTx, change, changeAddr, lastFee, nil
 }
 
+// ValidateAddress checks that the provided address is valid.
+func (dcr *ExchangeWallet) ValidateAddress(address string) bool {
+	_, err := stdaddr.DecodeAddress(address, dcr.chainParams)
+	return err == nil
+}
+
+// EstimateSendTxFee returns a tx fee estimate for sending or withdrawing the
+// provided amount using the provided feeRate.
+func (dcr *ExchangeWallet) EstimateSendTxFee(address string, sendAmount, feeRate uint64, subtract bool) (fee uint64, isValidAddress bool, err error) {
+	if sendAmount == 0 {
+		return 0, false, fmt.Errorf("cannot check fee: send amount = 0")
+	}
+
+	isValidAddress = dcr.ValidateAddress(address)
+
+	feeRate = dcr.feeRateWithFallback(feeRate)
+
+	// This is just an estimate so it is safe to ignore potential err.
+	addr, _ := stdaddr.DecodeAddress("DsTya4cCFBgtofDLiRhkyPYEQjgs3HnarVP", dcr.chainParams) // dummy P2PKHaddress
+
+	tx := wire.NewMsgTx()
+	payScriptVer, payScript := addr.PaymentScript()
```

Comment:
Here's the panic because `DecodeAddress` of that dummy fails on with simnet chain params it seems.
Looks like you need to update dcr's `EstimateSendTxFee` to work like btc's

Reply:
> Here's the panic because DecodeAddress of that dummy fails on with simnet chain params it seems.

I've updated dcr to use a dummy P2PKH script like btc. However, I'm unable to test with simnet(no setup yet). But it works fine for testnet.

Reply:
If you're setup for testnet, simnet is even easier:

``
alias dcrh='cd ~/github/decred/dcrdex/dex/testing/dcr && ./harness.sh'
alias btch='cd ~/github/decred/dcrdex/dex/testing/btc && ./harness.sh'
```
etc.

* start the above harnesses (`dcrh` etc. in separate terminals). NOTE: when you are done with these, use `./quit` in each
* wait for them to start and get to a prompt (done with bringup)
* start dcrdex server harness: `cd ~/github/decred/dcrdex/dex/testing/dcrdex && ./harness.sh' (DON'T use `quit` for this one, just exit/ctrl+d when done with it)
* build site bundle `alias sb='pushd ~/github/decred/dcrdex/client/webserver/site/ ; npm ci && npm run build ; popd'`
* build dexc
* nuke old dex.db and assetdb stuff `rm ~/.dexc/simnet/dexc.db ~/.dexc/simnet/backup/dexc.db && rm -r ~/.dexc/simnet/assetdb`
* `./dexc --simnnet`
* make wallets and send to them with `./sendtoaddress X N` from the relevant harness' terminal.

IF you accidentally detach or otherwise partially kill a harness, find it with `tmux ls` and attach with `tmux a {name}`

Reply:
@chappjc, thanks for the guide! Does simnet harness require me to run `dcrd --simnet` or `dcrwallet --simnet`?

Reply:
Those simnet harness scripts run all the dcrd and dcrwallet instances needed.

Reply:
I added `~/dextest` to my `$PATH` still get this error on tmux.
```
bash-3.2$ ./alpha sendmany default '{"SsjTp2QaT8qkPGWKKSeFi4DtaZsgkoPbgXt":24,"SsYW5LPmGCvvHuWok8U9FQu1kotv8LpvoEt":24,"Ssge52jCzbixgFC736RSTrwAnvH3a4hcPRX":24}'; tmux wait-for -S donedcr
open /Users/philemon/dextest/dcr/alpha/rpc.cert: no such file or directory
bash-3.2$ ./alpha sendmany default '{"SsjTp2QaT8qkPGWKKSeFi4DtaZsgkoPbgXt":24,"SsYW5LPmGCvvHuWok8U9FQu1kotv8LpvoEt":24,"Ssge52jCzbixgFC736RSTrwAnvH3a4hcPRX":24}'; tmux wait-for -S donedcr
open /Users/philemon/dextest/dcr/alpha/rpc.cert: no such file or directory
bash-3.2$ ./mine-alpha 2; tmux wait-for -S donedcr
open /Users/philemon/dextest/dcr/alpha/rpc.cert: no such file or directory
open /Users/philemon/dextest/dcr/alpha/rpc.cert: no such file or directory
```

Reply:
Don't put `~/dextest` on your path.  You just need to make sure that the binaries (dcrd, dcrwallet, bitcoind, etc.) are in a folder that is presently on your PATH.

Reply:
@chappjc, thanks. I'm set up for dcr simnet testing. Btw this guide could be somewhere on the `Simnet Testing` wiki page, right?

Reply:
I just do `rm -fr ~/.dexc/simnet/` to remove the simnet stuff. typing rm -fr is always scary though.

Reply:
> EDIT: Everythings look fine but the sent funds are not reflected.

Oh, sorry, I didn't mean `./sendtoaddress` I meant `./alpha sendtoaddress`.

Only on eth it's `./sendtoaddress`

Reply:
I don't know if am doing something wrong, but using `./alpha sendtoaddress ` just logs the txid in my console, and I still don't see the coins in my wallet.

Reply:
> Here's the panic because DecodeAddress of that dummy fails on with simnet chain params it seems.
Looks like you need to update dcr's EstimateSendTxFee to work like btc's

I can confirm dcr no longer have issues with simnet chain params.

Reply:
> EDIT: @chappjc, If you don't mind I could update the simnet wiki with the suggestions here.

What would you like to change on the wiki specifically?  I don't think it's a simnet specific thing so much as using the wrong chain parameters.